src/pool/srv_target.c

```diff
@@ -774,8 +775,15 @@ ds_pool_tgt_connect_handler(crt_rpc_t *rpc)
 	hdl->sph_capas = in->tci_capas;
 	hdl->sph_pool = pool;
 
+	rc = daos_iov_copy(&hdl->sph_cred, &in->tci_cred);
+	if (rc != 0) {
+		ds_pool_put(pool);
+		D_GOTO(out, rc);
+	}
+
 	rc = pool_hdl_add(hdl);
 	if (rc != 0) {
+		daos_iov_free(&hdl->sph_cred);
 		ds_pool_put(pool);
 		D_GOTO(out, rc);
```

Comment:
Not a fault of this PR, but D_FREE(hdl) here?

Reply:
I will make this fix as well and push a new update.